src/colorizer/FrameCache.ts

```diff
@@ -69,78 +73,65 @@ export default class FrameCache {
     this.setFirst(entry);
   }
 
-  public get length(): number {
-    return this.data.length;
+  public get size(): number {
+    return this.currentSize;
   }
 
-  /** Inserts a frame into the cache at `index` */
-  public insert(index: number, frame: Texture): void {
-    if (index >= this.data.length) {
-      return;
+  /**
+   * Inserts a value into the cache with the key. If the key already exists, the value is replaced.
+   * If the cache is full, the least recently used entry is evicted and its `dispose()` function is
+   * called if it exists.
+   *
+   * Throws an error if the size of the value is greater than the cache's maxSize.
+   */
+  public insert(key: string | number, value: E, size: number = 1): void {
```

Comment:
Variable size is useful if we want to define the cache's capacity in terms of bytes in the future.

Reply:
interesting how you did this.  I like it but I would suggest that sizes=1 are not very useful when they can range widely in how much memory they consume.

Reply:
Agreed! The `size=1` matches the previous behavior for frames, where the max capacity was defined in terms of the number of frames in the cache, but we should revisit this in the future. I'll add a TODO about it.